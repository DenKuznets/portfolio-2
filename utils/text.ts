const text = {
    showmore: 'Подробнее',
    backButton: 'Назад',
    myWorks: 'Мои проекты',
    allWorks: 'Все проекты',
    examples: 'Примеры работ',
    aboutMe: 'Обо мне',
    links: {
        inst: 'https://www.instagram.com/denis_kyznecov/',
        git: 'https://github.com/DenKuznets',
        tg: 'https://t.me/DenKuznets',
        project: '/projects/'
    },
    hero: {
        fname: 'Денис',
        lname: 'Кузнецов',
        profession: 'front-end разработчик',
        aboutme: 'верстаю сайты на react и styled-components',
        stack: 'СТЭК: react, nextjs, ts, mui, tailwind, styled-components, redux, rest, graphql, jest, react-testing-library, storybook',
        contactme: 'Связаться',
        socials: 'я в соц сетях'
    }
};

export default text;
