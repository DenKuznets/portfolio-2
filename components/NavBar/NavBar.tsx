import React from 'react';
import text from '../../utils/text';
import Link from 'next/link';

export interface NavBarProps extends React.HTMLAttributes<HTMLDivElement> {
    handleLinkClick?: () => void;
}

export const links = [
    { text: text.aboutMe, url: '#hero' },
    { text: text.examples, url: '#examples' }
];

const NavBar = ({className, handleLinkClick }: NavBarProps) => {
    return (
        <nav className={className}>
            <ul className="flex items-center justify-center gap-y-10 text-center font-serif max-md:min-h-screen max-md:flex-col max-md:text-[2rem] md:gap-x-16">
                {links.map((link) => (
                    <li key={link.url}>
                        <Link
                            className="text-gray-500 hover:text-almostBlack"
                            onClick={handleLinkClick}
                            href={link.url}
                        >
                            {link.text}
                        </Link>
                    </li>
                ))}
            </ul>
        </nav>
    );
};

export default NavBar;
