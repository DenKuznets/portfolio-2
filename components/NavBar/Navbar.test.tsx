import { render, screen } from "@testing-library/react";
import NavBar, { NAVBAR_TESTIDS } from "./NavBar";
import "@testing-library/jest-dom";
import text from "../../utils/text";

test("renders correctly", () => {
  render(<NavBar />);
  const container = screen.getByTestId(NAVBAR_TESTIDS.NAVBAR_CONTAINER);
  const list = screen.getByTestId(NAVBAR_TESTIDS.NAVBAR_LIST);
  const listItems = screen.getAllByTestId(NAVBAR_TESTIDS.NAVBAR_LIST_ITEM);
  expect(container).toBeInTheDocument();
  expect(list).toBeInTheDocument();
  expect(listItems).toHaveLength(Object.keys(text.header.nav).length);
});

test("renders navbar unchanged", () => {
  const { container } = render(<NavBar />);
  expect(container).toMatchSnapshot();
});
