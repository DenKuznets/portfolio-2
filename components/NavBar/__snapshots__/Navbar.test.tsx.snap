// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar unchanged 1`] = `
<div>
  <nav
    class="fixed md:static z-10 min-w-full md:min-w-fit bg-white/40 md:flex "
  >
    <ul
      class="flex-col w-[70vw] md:w-fit px-8 md:px-0 bg-white flex items-center justify-center gap-y-10 text-center font-serif min-h-screen md:min-w-0 md:min-h-fit md:shadow-none md:flex-row md:justify-end text-2xl md:text-xl shadow-lg md:gap-x-16 "
    >
      <li>
        <a
          class="text-gray-500 hover:text-almostBlack"
          href="/#hero"
        >
          Обо мне
        </a>
      </li>
      <li>
        <a
          class="text-gray-500 hover:text-almostBlack"
          href="/#examples"
        >
          Примеры работ
        </a>
      </li>
      <li>
        <a
          class="text-gray-500 hover:text-almostBlack"
          href="/projects"
        >
          Все проекты
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
